src/mailadm/bot.py

```diff
@@ -0,0 +1,124 @@
+import sys
+import sqlite3
+import time
+
+import deltachat
+from deltachat import account_hookimpl
+from mailadm.db import DB, get_db_path
+from mailadm.commands import add_user, add_token, list_tokens
+import os
+from threading import Event
+
+
+class SetupPlugin:
+    def __init__(self, admingrpid):
+        self.member_added = Event()
+        self.admingrpid = admingrpid
+        self.message_sent = Event()
+
+    @account_hookimpl
+    def ac_member_added(self, chat: deltachat.Chat, contact, actor, message):
+        if chat.id == self.admingrpid:
+            if chat.num_contacts() == 2:
+                while not contact.is_verified():
+                    time.sleep(0.1)
```

Comment:
why is this neccessary?  Because "member added can now happen before verification is complete"? That sounds wrong, though.  A member can only be added to a verified group if it is verified.  I'd consider this a bug and this is a dangerous work-around which can block the "event execution" overall for the bot if the contact ends up not getting verified (because of lost messages etc.).  If anything, then "for i in range(500): if contact.is_verified(): break else: time.sleep(...)" to limit the hanging.

Reply:
Ah, the reason is that sometimes the welcome message arrives at the user before the "verification complete" message arrives - this makes the welcome message unreadable to the user, as it is sent by an unverified contact to a verified group. This was ugly, and this small loop should ensure that the welcome message is at least *sent* after the last verification roundtrip message.

I will take it out for now, as it shouldn't be a blocker for this PR. Maybe delaying the welcome message 5 seconds would help.